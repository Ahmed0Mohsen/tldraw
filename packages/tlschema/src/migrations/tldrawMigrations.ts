import { MigrationSequence } from '@tldraw/store'
import { InitialMigration } from './000_InitialMigration'

/**
 * @public
 */
export const tldrawMigrations = {
	id: 'com.tldraw',
	migrations: [
		// New migrations must be appended AT THE END OF THIS ARRAY ONLY.
		InitialMigration,
	],
} as const satisfies MigrationSequence
