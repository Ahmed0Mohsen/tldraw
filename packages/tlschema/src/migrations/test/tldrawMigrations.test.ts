import { existsSync } from 'fs'
import { join } from 'path'
import { tldrawMigrations } from '../tldrawMigrations'

test('tldrawMigrations', () => {
	// It is EXTREMELY important that the order of the migrations is never changed after they have been run in production.
	// You can add to the end, but never remove or change the order of the migrations.
	expect(tldrawMigrations).toMatchInlineSnapshot(`
{
  "id": "com.tldraw",
  "migrations": [
    {
      "id": "com.tldraw/000_InitialMigration",
      "scope": "store",
      "up": [Function],
    },
  ],
}
`)
})

describe('each migration should be in a separate file', () => {
	const migrations = tldrawMigrations.migrations
	for (const migration of migrations) {
		const name = migration.id.split('/')[1]
		test(`src/migrations/${name}.ts should exist`, () => {
			expect(existsSync(join(__dirname, '..', name + '.ts'))).toBeTruthy()
		})
	}
})
