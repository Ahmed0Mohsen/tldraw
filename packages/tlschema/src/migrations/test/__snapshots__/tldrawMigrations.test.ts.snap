// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the order of migrations must not change over time 1`] = `
[
  {
    "id": "com.tldraw/000_InitialMigration",
    "scope": "store",
    "up": [Function],
  },
]
`;
